spaceMissionManager-service/src/main/java/cz/muni/fi/services/impl/MissionServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+                        "astronauts=" + mission.getAstronauts() +
+                        ", spacecrafts=" + mission.getSpacecrafts() +
+                        ", name='" + mission.getName() + '\'' +
+                        ", destination='" + mission.getDestination() + '\'' +
+                        ", eta=" + mission.getEta() +
+                        ", missionDescription='" + mission.getMissionDescription() + '\'' +
+                        ", endDate=" + mission.getEndDate() +
+                    '}');
+        updateMission(mission);
+    }
+
+    @Override
+    public void createMission(Mission mission) throws DataAccessException {
+        if (mission == null){
+            throw new IllegalArgumentException("Mission must not be null.");
+        }
+        try {
+            missionDao.createMission(mission);
+        } catch (Exception e) {
+            throw new ServiceDataAccessException("Could not create mission.", e);
+        }
+    }
+
+    @Override
+    public void cancelMission(Mission mission) throws DataAccessException {
+        if (mission == null){
+            throw new IllegalArgumentException("Mission must not be null.");
+        }
+        try {
+            missionDao.cancelMission(mission);
+        } catch (Exception e) {
+            throw new ServiceDataAccessException("Could not cancel mission.", e);
+        }
+    }
+
+    @Override
+    public List<Mission> findAllMissions() throws DataAccessException {
+        try {
+            return Collections.unmodifiableList(missionDao.findAllMissions());
+        } catch (Exception e) {
+            throw new ServiceDataAccessException("Could not find missions.", e);
+        }
+    }
+
+    @Override
+    public List<Mission> findAllMissions(boolean active) throws DataAccessException {
+        try {
+            return Collections.unmodifiableList(missionDao.findAllMissions(active));
+        } catch (Exception e) {
+            throw new ServiceDataAccessException("Could not find missions.", e);
+        }
+    }
+
+    @Override
+    public Mission findMissionById(Long id) throws DataAccessException {
+        if (id == null){
+            throw new IllegalArgumentException("Id must not be null.");
+        }
+        try {
+            return missionDao.findMissionById(id);
+        } catch (Exception e) {
+            throw new ServiceDataAccessException("Could not find mission.", e);
+        }
+    }
+
+    @Override
+    public void updateMission(Mission mission) throws DataAccessException {
+        if (mission == null){
+            throw new IllegalArgumentException("Mission must not be null.");
+        }
+        if(mission.isActive())
+            try {
+                missionDao.updateMission(mission);
+            } catch (Exception e) {
+                throw new ServiceDataAccessException("Could not update mission.", e);
+            }
+        else throw new IllegalArgumentException("Cannot modify archived mission");
```

Comment:
You have to check same mission in DB if that mission is not already archived (e.g you are trying to update archived mission and set mission.setActive(true))